src/locale/localed.c

```
@@ -395,6 +399,37 @@ static int method_set_locale(sd_bus_message *m, void *userdata, sd_bus_error *er
         for (LocaleVariable p = 0; p < _VARIABLE_LC_MAX; p++)
                 free_and_replace(c->locale[p], new_locale[p]);
 
+        /* Generate locale in case it is missing and the system is using locale-gen */
+        if (use_localegen) {
+                for (LocaleVariable p = 0; p < _VARIABLE_LC_MAX; p++) {
+                        if (isempty(c->locale[p]))
+                                continue;
+                        if (locale_is_installed(c->locale[p]))
+                                continue;
+
+                        r = locale_gen_enable_locale(c->locale[p]);
+                        if (r < 0) {
+                                if (r != -EINVAL) {
```

i figure this is so supposed to be ENOEXEC now?

Also, on ENOEXEC I figure we should just skip things entirely, i.e. jus finish the loop iteration, i.e.

```c
r = locale_gen_enable_locale(…);
if (r == -ENOEXEC)
        continue;
if (r < 0) {
        log_error_errno(r, …);
        return sd_bus_error_set_errnof(…);
}

r = locale_gen_run();
…
```

---

I think we should actually fail here now, in case we get an `-ENOEXEC` - because with the changes, getting that error code at that point means that the locale both isn't installed *and* can't be generated, so instead of skipping that problem we should likely error out with a better error message and have the user choose a different configuration (or fix their parameters).

---

(I changed the new code to fail, instead of skipping this event)